Applications/yOCTScanAndFindTissueSurface.m

```diff
@@ -1,44 +1,120 @@
+    xRange_mm, ...
+    yRange_mm, ...
+    'octProbePath', octProbePath, ...
+    'octProbeFOV_mm', octProbeFOV_mm, ...
+    'pixelSize_um', pixel_size_um, ...
+    'v',true  ...
+    );
+
+%% Reconstruct OCT Image for Subsequent Surface Analysis
+fprintf('Generating image...\n');
+outputTiffFile = [output_folder '\surface_analysis.tiff'];
+yOCTProcessTiledScan(...
+    volumeOutputFolder, ... Input
+    {outputTiffFile},... Save only Tiff file as folder will be generated after smoothing
+    'focusPositionInImageZpix',focusPositionInImageZpix,...
+    'dispersionQuadraticTerm',dispersionQuadraticTerm,... Use default
+    'cropZAroundFocusArea', false,...
+    'interpMethod','sinc5', ...
+    'v',true);
+[logMeanAbs, dimensions, ~] = yOCTFromTif(outputTiffFile);
+disp([newline, '-- Data loaded successfully.']);
 
 %% Estimate tissue surface
-% Todo
+fprintf('%s Identifying tissue surface...\n', datestr(datetime));
+startTimeSurfaceDetection = datetime; % Start timing the surface detection
+dimensions = yOCTChangeDimensionsStructureUnits(dimensions,'millimeters'); % Make sure it's in mm
+surfacePosition_mm = yOCTFindTissueSurface(logMeanAbs, dimensions, 'isVisualize', isVisualize);
+x_mm=dimensions.x.values;
+y_mm=dimensions.y.values;
+fprintf('Surface identification completed in %s.\n', datestr(datetime - startTimeSurfaceDetection, 'HH:MM:SS'));
 
 %% Clean up
-% delete temporary files
+if exist(output_folder, 'dir')
+    rmdir(output_folder, 's'); % Remove the output directory after processing
+end
+totalEndTime = datetime;  % Capture the ending time
+totalDuration = totalEndTime - totalStartTime;
+fprintf('Total surface analysis process completed in %s.\n', datestr(totalDuration, 'HH:MM:SS'));
 
-%% Return
-% Tbd, this is just a demo
-surfacePosition_mm = ones(10,10);
-x_mm=1:10;
-y_mm=1:10;
+%% Check if the average surface distance is within the acceptable range
+average_surface_distance = nanmean(nanmean(surfacePosition_mm)); % Calculate the average surface distance
+acceptableRange_mm = 0.01; % 10 microns is acceptable range
+if isnan(average_surface_distance)
+    warning(['No tissue identification possible. Likely the tissue is out of focus below the detection range. ' ...
+        'Please manually adjust the stage UP to bring the tissue into focus.']);
+    isSurfaceFocused = false; % Unable to verify if tissue surface is at focus
+elseif abs(average_surface_distance) > acceptableRange_mm
+    incrementsToAdjust = round(abs(average_surface_distance) / acceptableRange_mm); % Number of increments to adjust
+    if average_surface_distance > 0 % Determine direction of adjustment
+        direction = 'UP';
+    else
+        direction = 'DOWN';
+    end
+    warning('off', 'backtrace'); % Temporarily turn off backtrace for clear messaging
+    warning(sprintf(['The average distance of the surface (%.4f mm) is out of range.\n\n', ...
+        'Move the stage %s by %.2f mm to bring the tissue surface into focus. \n'], ...
```

Comment:
Up/Down might be a little confusing. Do you mean increase stage Z position = Up?
You can say "increase z by x"